### DISCUSSION TITLE
How best to implement header and footers reports

### DISCUSSION THREAD

**Discussion by vforget** [datapane/datapane#380] [Help]
### Summary

Hi all, 

What would be the best way to insert a company image header on a report? I used `dp.Media()` and it works. However, the documentation mentions to use `dp.HTML()` for company logos, etc, but the image I link to in the <img/> object in the html code doesn't get embedded into the HTML when I run `dp.save_report`.

So, what is the preferred way to add header or footers to a report?

Reference: https://docs.datapane.com/reference/display-blocks/text/#simple-latex-formula:~:text=company%27s%20brand%2C%20logo%2C%20and%20more

### Additional information

```js
import altair as alt
import datapane as dp
from vega_datasets import data

html = """
<img src="./header_1.png" alt="5 Prime Sciences Header">
"""
view = dp.HTML(html)

dp.save_report(view, "quickstart_report.html", open=True)
```

**Comment by Benrdn** [comment]
Maybe the documentation is referring to converting the logo into html.
For example for an animated gif I use this:

```
<!DOCTYPE html>
<html>
<head>
  <title>Animated GIF</title>
  <style>
    img {
      float: right;
    }
  </style>
</head>
<body>
  <img src="data:image/gif;base64,R0lGODlhAQABAIAAAAUEBAAAACwAAAAAAQABAAACAkQBADs=" alt="Animated GIF" style="width: 300px;">
  <script>
    window.onload = function() {
      var img = document.querySelector('img');
      img.src = "data:image/gif;base64," +
        "YOUR_BASE64_ENCODED_IMAGE_DATA_HERE";
    }
  </script>
</body>
</html>
```
where "YOUR_BASE64_ENCODED_IMAGE_DATA_HERE" is the BASE64 data I got from an online converter from gif to BASE64

**Reply by mands** [reply]
That's an interesting solution! :) Does `dp.Media` not support animated GIFs - if not do let us know and we will push an update.

**Comment by mands** [comment]
Hi @vforget - apologies for the delay! For your use-case I'd recommend using `dp.Media`, as it will automatically include the image file itself.

`dp.HTML` should be used for more advanced layouts that you want to use/reuse in your report, such as a logo with clickable links, etc..
